lib/tasks/audits.rake

```diff
@@ -21,4 +22,12 @@ namespace :audits do
     count = audits.delete_all
     puts "Successfully deleted #{count} audits!"
   end
+
+  task :anonymize => :environment do
+    before = ENV['days'].to_i.days.ago if ENV['days']
+    before ||= 90.days.ago
+    count = Audit.where('created_at <= ?', before).where.not(:remote_address => nil, :user_id => nil, :username => nil)
```

Comment:
You can use the `Audit.up_until` scope here like in the other task. Perhaps combine the first part of the tasks to a function that finds the audits and then the task itself only has the difference between the two. 
Also, might be faster to skip the additional condition since that will require a possibly slow lookup to find all audits that don't match the condition yet and just nullify everything older then the date - but it might be a good idea to benchmark which is faster.

Reply:
It is slightly faster with the additional condition. At 1.5 million audits up_until(before) with 4000 to anonymise, it took 3.7s with the condition and 76.6s without. This obviously will be different when you have to anonymise a higher percentage of the audits up_until(before), but if you run the task regularly then this implementation should be advantages.
I will refactor the tasks like you suggested.